packages/ember-data/lib/serializers/rest-serializer.js

```diff
@@ -40,8 +40,12 @@ function coerceId(id) {
 
   ```js
   App.ApplicationSerializer = DS.RESTSerializer.extend({
-    keyForAttribute: function(attr) {
-      return Ember.String.underscore(attr).toUpperCase();
+    keyForAttribute: function(attr, method) {
+      if (method === 'serialize') {
+        return Ember.String.underscore(attr).toUpperCase();
+      } else {
+        return Ember.String.camelize(attr.toLowerCase());
+      }
```

Comment:
Lets leave this example as it was (although feel free to add the `method` parameter to the function declaration). The new implementation doesn't match the text above it.